skbio/sequence/_sequence.py

```diff
@@ -1302,16 +1432,21 @@ def _copy(self, deep, memo):
         bytes_ = np.copy(self._bytes)
 
         seq_copy = self._constructor(sequence=bytes_, metadata=None,
-                                     positional_metadata=None)
+                                     positional_metadata=None,
+                                     interval_metadata=None)
 
         if deep:
             seq_copy._metadata = MetadataMixin._deepcopy_(self, memo)
             seq_copy._positional_metadata = \
                 PositionalMetadataMixin._deepcopy_(self, memo)
+            seq_copy._interval_metadata = IntervalMetadataMixin._deepcopy_(
```

Comment:
Its the way that the API is setup at the moment.  Its because the metadata objects have their own copy methods.  So a check needs to be done to see if the metadata exists in the first place, before calling the copy command.  I don't have a strong reference for replacing this, though keep in mind that will require refactoring a bit of legacy code.

@ebolyen do you think it is worthwhile to do that refactor here?  Or is it better to submit an issue and address this down the road?